### ISSUE TITLE
Order of scheduler/optimizer calls

### ISSUE THREAD

**Issue by RuslanSel** [NVIDIA/NeMo#7660]
I have the same problem as in #6869 with punctuation_capitalization_train_evaluate.py

UserWarning: Detected call of lr_scheduler.step() before optimizer.step(). In PyTorch 1.1.0 and later, you should call them in the opposite order: optimizer.step() before lr_scheduler.step(). Failure to do this will result in PyTorch skipping the first value of the learning rate schedule.
Error executing job with overrides

NeMo: 1.21.0
OS version: Ubuntu 18.04
PyTorch version: '2.1.0+cu121'
Python version: Python 3.11.5

**Comment by titu1994** [comment]
This happens potential under ddp and is not practically avoidable. It has no significant effects on training.
